catalog/src/main/java/org/killbill/billing/catalog/api/user/DefaultCatalogUserApi.java

```diff
@@ -224,5 +203,34 @@ private InternalTenantContext createInternalTenantContext(final TenantContext te
         return internalCallContextFactory.createInternalTenantContextWithoutAccountRecordId(tenantContext);
     }
 
+    private ValidationErrors validateCatalogInternal(final String catalogXML, final InternalTenantContext internalTenantContext) {
+        final ValidationErrors errors = new ValidationErrors();
+        try {
+            VersionedCatalog versionedCatalog = catalogService.getFullCatalog(false, true, internalTenantContext);
+            if (versionedCatalog == null) {
+                versionedCatalog = new DefaultVersionedCatalog();
+            }
+            // Validation purpose:  Will throw if bad XML or catalog validation fails
+            final InputStream stream = new ByteArrayInputStream(catalogXML.getBytes());
+            final StaticCatalog newCatalogVersion = XMLLoader.getObjectFromStream(stream, StandaloneCatalog.class);
+
+            if (versionedCatalog.getCatalogName() != null && !versionedCatalog.getCatalogName().isEmpty() && !newCatalogVersion.getCatalogName().equals(versionedCatalog.getCatalogName())) {
+                errors.add(new ValidationError(String.format("Catalog name '%s' is different from existing catalog name '%s'", newCatalogVersion.getCatalogName(), versionedCatalog.getCatalogName()), StaticCatalog.class, "")); //TODO_1674 - Error message is hardcoded here, move this elsewhere?
+                return errors;
+            }
 
+            ((DefaultVersionedCatalog) versionedCatalog).add((StandaloneCatalog) newCatalogVersion);
+            ((DefaultVersionedCatalog) versionedCatalog).validate(null, errors);
+
+        } catch (final CatalogApiException e) {
+            errors.add(new ValidationError(e.getMessage(), DefaultVersionedCatalog.class, ""));
+        } catch (final ValidationException e) {
+            errors.addAll(e.getErrors());
+        } catch (final JAXBException e) {
+            errors.add(new ValidationError(e.getLinkedException().getMessage(), DefaultVersionedCatalog.class, ""));
```

Comment:
Maybe NPE check on `e.getLinkedException()` just in case?

Reply:
Added a null check for `e.getLinkedException()` LMK if this is what you were referring to.

Reply:
I don't see it, maybe the PR was merged too early?

Reply:
Yes, the old PR was merged so I've created a new PR: https://github.com/killbill/killbill/pull/1813